### DISCUSSION TITLE
Prop to prevent QSelect from closing when searching

### DISCUSSION THREAD

**Discussion by pikax** [quasarframework/quasar#13969] [Ideas / Proposals]
When  using QSelect with a `search`, every time the search happens the QMenu is closed until we resolve the `search` items, this UX is nice but it creates popin/popout of the menu, to prevent that I suggest adding a new `prop` to QSelect to prevent that from happening.

### Reproduction
Check [lazy filtering](https://quasar.dev/vue-components/select#example--lazy-filtering), just type something wait until it completes and type again.

### Desire behaviour

Similar to how "google behaves", you type something and we show it, if you keep typing it will fetch new results but it will keep the menu open.

<img width="639" alt="image" src="https://user-images.githubusercontent.com/4620458/179212207-38310a85-1fbd-452c-8728-137c5fd08135.png">

### Other ways

This can be achieved on the user-land by not using QSelect, but it would be nice to have an opt-in way to get this behaviour. 

### Possible issues

One issue I see is the user feedback, because the menu is always present it might confuse some users, but I believe the spinner on the input is enough for showing the QSelect is fetching data.

## Suggested change

Property name maybe requires a renaming

```diff
diff --git a/src/components/select/QSelect.js b/src/components/select/QSelect.js
index 4b39e61afd10025040b3776372d1f8fb9dc78eb8..60de39bc5ab6e872e5cf533c0c1fd96f7b1d1697 100644
--- a/src/components/select/QSelect.js
+++ b/src/components/select/QSelect.js
@@ -122,7 +122,9 @@ export default createComponent({
     },
 
     onNewValue: Function,
-    onFilter: Function
+    onFilter: Function,
+
+    keepMenuOnSearch: Boolean
   },
 
   emits: [
@@ -1089,7 +1091,7 @@ export default createComponent({
       }
 
       const localFilterId = setTimeout(() => {
-        menu.value === true && (menu.value = false)
+        !props.keepMenuOnSearch && menu.value === true && (menu.value = false)
       }, 10)
 
       clearTimeout(filterId)
@@ -1133,7 +1135,7 @@ export default createComponent({
             state.innerLoading.value = false
             innerLoadingIndicator.value = false
           }
-          menu.value === true && (menu.value = false)
+          !props.keepMenuOnSearch && menu.value === true && (menu.value = false)
         }
       )
     }
```

**Comment by pdanpdan** [comment]
There is no need for a new prop - there is a trick :)
https://codepen.io/pdanpdan/pen/WNJXNzw?editors=0010

call `update()` once at start of filter function and then call it again when you have your data
but you need to manage the loading indicator
